File: p25304.js
const fs = require('fs') 
const inputData = fs.readFileSync('dev/stdin').toString().split('\n')


let sum = 0
for(let i=2;i<=+inputData[1]+1;i++){
    const splited = inputData[i].split(' ').map(e=>+e)  
    sum += splited[0] * splited[1]
}
+inputData[0] === sum ? "Yes" :"No"


console.log(+inputData[0] === sum ? "Yes" :"No")

File: cpp/p25304.cpp
#include <iostream>
using namespace std;

int main(){
    int X , N;
    cin >> X >> N;
    int sum = 0;
    for(int i=0;i<N;i++){
        int a,b;
        cin >> a>> b;
        sum += a*b;
    }
    if(sum==X) cout << "Yes";
    else cout << "No";
    return 0;
}